### PR TITLE
Sync GHA setup with CCCL repo

### PR DESCRIPTION
**Pull request by sleeepyjack**
This PR applies the latest updates/features from the CCCL GHA setup.

Superseeds #355

### CODE REVIEW

File: ci/matrix.yml [hunk 1]
```diff
@@ -41,6 +42,6 @@ devcontainer_version: '23.08'
 # Configurations that will run for every PR
 pull_request:
   nvcc:
-    # There is currently only one CUDA 11.8 image available which comes with the system's default C++ compiler. For ubuntu22.04, we know that the default CC is gcc11.3
-    - {cuda: *cuda_oldest, os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'gcc', version: '11', exe: 'c++'}, gpu_build_archs: '60', std: [17], jobs: ['build', 'test']}
-    - {cuda: *cuda_newest, os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'gcc', version: '12', exe: 'g++'}, gpu_build_archs: '70', std: [17], jobs: ['build', 'test']}
\ No newline at end of file
+    - {cuda: *cuda_oldest, os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'gcc', version: '11', exe: 'g++'}, gpu_build_archs: '70', std: [17], jobs: ['build', 'test']}
+    - {cuda: *cuda_newest, os: 'ubuntu22.04', cpu: 'amd64', compiler: {name: 'gcc', version: '12', exe: 'g++'}, gpu_build_archs: '70', std: [17], jobs: ['build', 'test']}
+    - {cuda: *cuda_newest, os: 'ubuntu22.04', cpu: 'arm64', compiler: {name: 'gcc', version: '12', exe: 'g++'}, gpu_build_archs: '60,90', std: [17], jobs: ['build']}
```
Comment (PointKernel) [inline ci/matrix.yml:47]: Can we also test this?

**Review by PointKernel — APPROVED** [review]
Me mostly learning what's going on instead of reviewing.

Not sure my approval should count.

File: ci/matrix.yml (hunk 1 above)
Comment (sleeepyjack) [reply ci/matrix.yml:47]: I think we don't have arm64 GPU runners atm but let me check.

**Comment by sleeepyjack** [comment]
@PointKernel thanks!
I'll merge the PR so I can get started with the `clang-cuda` build test.

File: ci/matrix.yml (hunk 1 above)
Comment (sleeepyjack) [reply ci/matrix.yml:47]: Yep, no arm64 gpu runners yet

**Pull request merged**